src/txn/txn_timestamp.c

```diff
@@ -867,6 +867,49 @@ __wt_txn_set_read_timestamp(WT_SESSION_IMPL *session, wt_timestamp_t read_ts)
     return (0);
 }
 
+/*
+ * __wt_txn_set_timestamp_internal --
+ *     Does the "heavy lifting" for the outside-facing txn_set_timestamp functions.
+ */
+static int
+__wt_txn_set_timestamp_internal(
+  WT_SESSION_IMPL *session, uint64_t commit, uint64_t durable, uint64_t prepare, uint64_t read)
+{
+    WT_CONNECTION_IMPL *conn;
+    bool set_ts;
+
+    conn = S2C(session);
+    set_ts =
+      commit != WT_TS_NONE || durable != WT_TS_NONE || prepare != WT_TS_NONE || read != WT_TS_NONE;
+
+    /* Look for a commit timestamp. */
+    if (commit != WT_TS_NONE)
+        WT_RET(__wt_txn_set_commit_timestamp(session, commit));
+
+    /*
+     * Look for a durable timestamp. Durable timestamp should be set only after setting the commit
+     * timestamp.
+     */
+    if (durable != WT_TS_NONE)
+        WT_RET(__wt_txn_set_durable_timestamp(session, durable));
+    __wt_txn_publish_durable_timestamp(session);
```

Comment:
Interestingly, it wouldn't get updated to durable in that case. Since the transaction would already have `WT_TXN_SHARED_TS_DURABLE` set, it would leave the published `commit` timestamp as the shared one.

Reply:
Yes, @agorrod you are correct. The `durable timestamp` is not published again once the `commit timestamp` is published.